### PR TITLE
LB-603: Fix spark crash on calculation of all_time release stats

### PR DESCRIPTION
**Pull request by ishaanshah**
# Problem
The spark cluster crashes when calculating all_time release stats because of memory issues.

# Solution
The `collect` command collects all the data and dumps it into the driver, hence running out of memory. We should use `toLocalIterator` to iterate over the instead of collecting it at once.

# Action
We should test if the updated code works correctly without crashing on production. The flag `--all-time-release` can be used to trigger all_time release stats calculation only.

### CODE REVIEW

File: listenbrainz/webserver/views/stats_api.py [hunk 1]
```diff
@@ -237,7 +237,7 @@ def _process_entity(user_name, stats, stats_range, offset, count, entity):
     count = min(count, MAX_ITEMS_PER_GET)
     try:
         total_entity_count = stats[entity][stats_range]['count']
-    except KeyError:
+    except (TypeError, KeyError):
```
Comment (ishaanshah) [inline listenbrainz/webserver/views/stats_api.py:240]: I will add a test for this in a subsequent PR

**Comment by paramsingh** [comment]
Running this in prod.

**Comment by paramsingh** [comment]
This is going reasonably well, as far as I can see. There are around 4.8k messages in the rabbitmq queue right now, so the processing worked, i'm gonna wait until it reports completion.

File: listenbrainz/webserver/views/stats_api.py (hunk 1 above)
Comment (ishaanshah) [reply listenbrainz/webserver/views/stats_api.py:240]: Added in #898

**Pull request merged**